client/src/test/java/io/spine/core/CommandsTest.java

```diff
@@ -192,49 +193,47 @@ void createWereBetweenPredicate() {
                     requestFactory.createCommand(BoolValue.getDefaultInstance(),
                                                  secondsAgo(5));
 
-            ImmutableList<Command> commands =
-                    ImmutableList.of(fiveMinsAgo, twoMinsAgo, thirtySecondsAgo, twentySecondsAgo,
-                                     fiveSecondsAgo);
-            Iterable<Command> filter = commands.stream()
-                                               .filter(Commands.wereWithinPeriod(
-                                                       minutesAgo(3), secondsAgo(10))::apply)
-                                               .collect(toList());
+            Iterable<Command> filter = Stream.of(fiveMinsAgo,
+                                                 twoMinsAgo,
+                                                 thirtySecondsAgo,
+                                                 twentySecondsAgo,
+                                                 fiveSecondsAgo)
+                                             .filter(wereWithinPeriod(minutesAgo(3), secondsAgo(10))
+                                                             ::apply)
+                                             .collect(toList());
 
             assertEquals(3, StreamSupport.stream(filter.spliterator(), false)
                                          .count());
```

Comment:
This is still strange.

First, we generate a stream, then convert it to list and then back to stream, and, at last, obtain `count()`.